src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -115,12 +124,33 @@ public bool ExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionConte
         }
 
         private void CodeCleanupOrFormat(
-            FormatDocumentCommandArgs args, Document document,
+            FormatDocumentCommandArgs args, Document document, ICodeCleanupService codeCleanupService,
             IProgressTracker progressTracker, CancellationToken cancellationToken)
         {
-            var codeCleanupService = document.GetLanguageService<ICodeCleanupService>();
-
             var docOptions = document.GetOptionsAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+            var workspace = document.Project.Solution.Workspace;
+
+            // AB Test
+            // If the feature is OFF and the feature options have not yet been set to their assigned flight, do so now
+            if (!docOptions.GetOption(CodeCleanupOptions.PerformAdditionalCodeCleanupDuringFormatting)
```

Comment:
why do you need to get this info from doc option? can user set this option from editorconfig file? I thought it only exist in tools->option page?